src/node_buffer.cc

```diff
@@ -19,7 +19,7 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Local<Object> obj = argT.As<Object>();                                    \
```

Comment:
can't exclude the proper checks from C++. example:
```javascript
function F() { }
F.prototype.__proto__ = Buffer.prototype;
var f = new F();

f.write();  // ABORT
```
The above passes the `instanceof` check. The only think you can ensure is that it's an `Object`, but no that it's been allocated external memory. Still run the `Buffer::HasInstance()` check on all incoming objects.